source/agora/stats/Server.d

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+
+    Starts up a HTTP server and listen to queries from Prometheus
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.stats.Server;
+
+import agora.stats.Utils;
+
+import ocean.util.prometheus.collector.CollectorRegistry;
+import vibe.vibe;
+
+class StatsServer
+{
+    private HTTPListener http_listener;
+
+    /***************************************************************************
+
+            Constructs a StatsServer instance
+
+            Params
+                port = port on which the server will listen on,
+                -1 to disable listening
+
+    ***************************************************************************/
+
+    public this (int port)
+    {
+        if (port == -1)
```

Comment:
if this triggers anyone, I can create a MockStatsServer that does nothing and use it for the unit tests